core/FoundDefinitions.h

```diff
@@ -82,43 +78,33 @@ class FoundDefinitionRef final {
     FoundField &field(FoundDefinitions &foundDefs);
     const FoundField &field(const FoundDefinitions &foundDefs) const;
 
-    core::SymbolRef symbol() const;
+    core::ClassOrModuleRef symbol() const;
 
     static std::string kindToString(Kind kind);
 
     std::string toString(const core::GlobalState &gs, const FoundDefinitions &foundDefs) const;
 };
 CheckSize(FoundDefinitionRef, 4, 4);
 
-struct FoundClassRef final {
-    core::NameRef name;
-    core::LocOffsets loc;
-    // If !owner.exists(), owner is determined by reference site.
-    FoundDefinitionRef owner;
-
-    std::string toString(const core::GlobalState &gs, const FoundDefinitions &foundDefs, uint32_t id) const;
-};
-CheckSize(FoundClassRef, 16, 4);
-
 struct FoundClass final {
     FoundDefinitionRef owner;
-    FoundDefinitionRef klass;
+    core::NameRef name;
     core::LocOffsets loc;
     core::LocOffsets declLoc;
 
     enum class Kind : uint8_t {
+        Unknown,
         Module,
         Class,
     };
     Kind classKind;
 
     std::string toString(const core::GlobalState &gs, const FoundDefinitions &foundDefs, uint32_t id) const;
 };
-CheckSize(FoundClass, 28, 4);
+CheckSize(FoundClass, 24, 4);
```

Comment:
I am surprised at this change...`FoundDefinitionRef` and `NameRef` are the same size...so this structure shouldn't change size?

I could see `FoundStaticField` shrinking, though.

Reply:
good point, and the build failed actually